sdk/core/azure-core/src/main/java/com/azure/core/util/ClientOptions.java

```diff
@@ -0,0 +1,168 @@
+     *
+     * @param name the name
+     * @param value the value
+     * @return The updated ClientOptions object
+     *
+     * @throws NullPointerException if {@code name} or {@code value} is null.
+     */
+    public ClientOptions addHeader(String name, String value) {
+        Objects.requireNonNull(name, "'name' cannot be null.");
+        Objects.requireNonNull(value, "'value' cannot be null.");
+
+        Header existing = get(name);
+        if (existing == null) {
+            this.headers.put(formatKey(name), new Header(name, value));
+        } else {
+            existing.addValue(value);
+        }
+        return this;
+    }
+
+    /**
+     * Gets the {@link Header header} for the provided header name. {@code Null} is returned if the header isn't
+     * found.
+     *
+     * @param name the name of the header to find.
+     * @return the header if found, null otherwise.
+     *
+     * @throws NullPointerException if {@code name} is null.
+     */
+    public Header get(String name) {
+        Objects.requireNonNull(name, "'name' cannot be null.");
+
+        return headers.get(formatKey(name));
+    }
+
+    /**
+     * Removes the {@link Header header} with the provided header name. {@code Null} is returned if the header
+     * isn't found.
+     *
+     * @param name the name of the header to remove.
+     * @return the header if removed, null otherwise.
+     *
+     * @throws NullPointerException if {@code name} is null.
+     */
+    public Header remove(String name) {
+        Objects.requireNonNull(name, "'name' cannot be null.");
+
+        return headers.remove(formatKey(name));
+    }
+
+    /**
+     * Get the value for the provided header name. {@code Null} is returned if the header name isn't found.
+     *
+     * @param name the name of the header whose value is being retrieved.
+     * @return the value of the header, or null if the header isn't found
+     *
+     * @throws NullPointerException if {@code name} is null.
+     */
+    public String getValue(String name) {
+        final Header header = get(name);
+        return header == null ? null : header.getValue();
+    }
+
+    /**
+     * Get the values for the provided header name. {@code Null} is returned if the header name isn't found.
+     *
+     * <p>This returns {@link #getValue(String) getValue} split by {@code comma}.</p>
+     *
+     * @param name the name of the header whose value is being retrieved.
+     * @return the values of the header, or null if the header isn't found
+     */
+    public String[] getValues(String name) {
+        final Header header = get(name);
+        return header == null ? null : header.getValues();
+    }
+
+    /**
+     * Gets a {@link Map} representation of the {@link Header} collection. The keys are {@link Header#getName()} in the
+     * given {@link Header}.
+     *
+     * @return the headers as map
+     */
+    public Map<String, Header> toMap() {
+        return headers;
```

Comment:
I don't think we need any of these methods on the `ClientOptions` class. We should just have getter method for returning the header collection.